index.rb

```diff
@@ -1,160 +1,262 @@
+      end
+
+      # If just a specific character is given
+      my_all_boolean = true
+      each do |i|
+        my_all_boolean = false unless i == args.first
+      end
+      return my_all_boolean
     end
-    my_all_boolean = true
-    for i in self
 
-      if !yield(i)
-        my_all_boolean = false
+    # Checks for when a block is not given
+    unless block_given?
+      my_all_boolean = true
+      each do |i|
+        my_all_boolean = false unless i
       end
+      return my_all_boolean
+    end
 
+    # If a block is given
+    my_all_boolean = true
+    each do |i|
+      my_all_boolean = false unless yield(i)
     end
-    puts my_all_boolean
+    my_all_boolean
   end
 
-  def my_any?
-    if self.length == 0 then
-      return false
+  def my_any?(*args)
+    if args.length == 1 && !block_given?
+
+      if args.first.instance_of?(Regexp)
+        my_any_boolean = false
+        each do |i|
+          my_any_boolean = true if args.first.match(i)
+        end
+        return my_any_boolean
+      end
+
+      if args.first.instance_of?(Class)
+        my_any_boolean = false
+        each do |i|
+          my_any_boolean = true if i.instance_of?(args.first)
+        end
+        return my_any_boolean
+      end
+
+      my_any_boolean = false
+      each do |i|
+        my_any_boolean = true if i == args.first
+      end
+      return my_any_boolean
+    end
+    unless block_given?
+      my_any_boolean = false
+      each do |i|
+        my_any_boolean = true if i
+      end
+      return my_any_boolean
     end
     my_any_boolean = false
+    each do |i|
+      my_any_boolean = true if yield(i)
+    end
+    my_any_boolean
+  end
 
-    for i in self
+  def my_none?(*args)
+    if args.length == 1 && !block_given?
 
-      if yield(i)
-        my_any_boolean = true
+      if args.first.instance_of?(Regexp)
+        my_none_boolean = true
+        each do |i|
+          my_none_boolean = false if args.first.match(i)
+        end
+        return my_none_boolean
       end
 
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i == args.first
+      end
+      return my_none_boolean
     end
-    puts my_any_boolean
-  end
 
-  def my_none?
-    if self.length == 0 then
-      return true
+    unless block_given?
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i
+      end
+      return my_none_boolean
     end
-    my_none_boolean = true
-    for i in self
 
-      if yield(i)
-        my_none_boolean = false
+    if args.length == 1
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i.is_a?(args.first)
       end
+      return my_none_boolean
+
+    end
 
+    my_none_boolean = true
+    each do |i|
+      my_none_boolean = false if yield(i)
     end
-    puts my_none_boolean
+    my_none_boolean
   end
 
+  def my_inject(*args)
+    return my_inject { |n, i| n.send(args.first, i) } if args.length == 1 && !block_given?
 
```

Comment:
- [ ] my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block. that means it has to accept blocks with operation like { |sum, n| sum + n }  and return the correct answer.
- [ ] my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method. that means it has to accept a symbol like this [1,2].my_inject(:+). and this should combine all elements.

Reply:
My inject works with both of the test cases mentioned above, but reviewers are saying it doesn't, I don't get why that is.